app/src/main/java/com/piwniczna/mojakancelaria/activities/LoginActivity.kt

```diff
@@ -24,47 +29,58 @@ class LoginActivity : AppCompatActivity() {
         passwordEditText = findViewById(R.id.passwordCodeEditText)
         loginButton = findViewById(R.id.loginButton)
 
-        testDb()
+        initDb(this)
     }
 
     fun handleLogin(view: View) {
-        // Log.e("Login","Jestem")
-        // TODO: check password
+        val pin = passwordEditText.text
+        val pinHash = Hashing.sha256()
+                .hashString(pin, StandardCharsets.UTF_8)
+                .toString()
+
+        AsyncTask.execute {
+            val dbHash = database.dao().getHash()
+
+            runOnUiThread {
+
+                if (!pinHash.equals(dbHash.hash)) {
+                    val text = R.string.loginFailedToast
```

Comment:
Are these variables necessary?
https://techbeacon.com/app-dev-testing/why-unnecessary-variables-are-bad-your-code

Reply:
We can instead of using this values (not variables!) make one long and ugly function call...

Reply:
meh